internal/langserver/handlers/did_change.go

```diff
@@ -82,7 +84,16 @@ func TextDocumentDidChange(ctx context.Context, params lsp.DidChangeTextDocument
 	if err != nil {
 		return err
 	}
-	diags.PublishHCLDiags(ctx, mod.Path, mod.Diagnostics, "HCL")
+
+	d := map[string]hcl.Diagnostics{}
+	for _, md := range []interface{}{mod.ConfigDiagnostics, mod.VarsDiagnostics} {
+		err = mergo.Merge(&d, md)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
this should probably be a function of the `state.Module`; however, since `module.Module` is an alias to the previous one, I don't seem to be able to call it:

```
func (m *Module) Diagnostics() (map[]hcl.Diagnostics, error) {
          d := map[string]hcl.Diagnostics{}	
          for _, md := range []interface{}{mod.ConfigDiagnostics, mod.VarsDiagnostics} {		
                  err = mergo.Merge(&amp;d, md)		
                  if err != nil {			
                         return d, err		
                  }	
              }
         return d, nil
}
```